File: Palindrome SubStrings/Palindrome_SubStrings.py
class Solution:

    def countPS(self, s):
        ans = 0
        n = len(s)
        for i in range(0,n):
            left = i
            right = i
            # for odd length palindrome substring
            while left >=0 and right < n and s[left] == s[right]:
                if right-left+1 >= 2:
                    ans +=1
                left -=1
                right +=1
            # for even length palindrome substring
            left = i;
            right = i+1;
            while left >=0 and right < n and s[left] == s[right]:
                if right-left+1 >= 2:
                    ans +=1
                left -=1
                right +=1
        
        return ans;

File: Palindrome SubStrings/Palindrome_SubStrings.cpp
class Solution {
  public:
    int countPS(string &s) {
        int ans = 0;
        int n = s.size();
        for(int i = 0; i<n; i++) {
            int left = i,right = i;
            // for odd length palindrome substring
            while(left >=0 && right < n && s[left] == s[right]) {
                if(right-left+1 >= 2) {
                    ans++;
                } 
                left--;
                right++;
            }
            // for even length palindrome substring
            left = i;
            right = i+1;
            while(left >= 0 && right < n && s[left] == s[right]) {
                if(right-left+1 >=2) {
                    ans++;
                }
                left--;
                right++;
            }
        }
        return ans;
    }
};